zebra/zebra_vty.c

```diff
@@ -1020,65 +1013,87 @@ DEFUN (show_ipv6_nht_vrf_all,
 	return CMD_SUCCESS;
 }
 
-DEFUN (ip_nht_default_route,
+DEFPY (ip_nht_default_route,
        ip_nht_default_route_cmd,
        "ip nht resolve-via-default",
        IP_STR
        "Filter Next Hop tracking route resolution\n"
        "Resolve via default route\n")
 {
+	ZEBRA_DECLVAR_CONTEXT(vrf, zvrf);
+
+	if (!zvrf)
+		return CMD_WARNING;
+
 	if (zebra_rnh_ip_default_route)
 		return CMD_SUCCESS;
 
 	zebra_rnh_ip_default_route = 1;
-	zebra_evaluate_rnh(VRF_DEFAULT, AF_INET, 1, RNH_NEXTHOP_TYPE, NULL);
+
+	zebra_evaluate_rnh(zvrf, AF_INET, 1, RNH_NEXTHOP_TYPE, NULL);
 	return CMD_SUCCESS;
 }
 
-DEFUN (no_ip_nht_default_route,
+DEFPY (no_ip_nht_default_route,
```

Comment:
Again no n3eed to change from DEFUN to DEFPY no feature taken advantage of here.